src/TurbulenceClosures/turbulence_closure_implementations/CATKEVerticalDiffusivities/turbulent_kinetic_energy_equation.jl

```diff
@@ -94,3 +169,159 @@ end
+
+    return _top_tke_flux(i, j, grid, tke_parameters, closure,
+                         buoyancy, fields, top_tracer_bcs, top_velocity_bcs, clock)
+end
+
+""" Compute the flux of TKE through the surface / top boundary. """
+@inline top_tke_flux(i, j, grid, clock, fields, parameters, closure, buoyancy) = zero(grid)
+@inline inner_top_tke_flux(i, j, grid, clock, fields, parameters, closure, buoyancy) = zero(grid)
+
+#=
+@inline inner_top_tke_flux(i, j, grid, clock, fields, parameters, closure::Tuple{}, buoyancy) = zero(grid)
+@inline top_tke_flux(i, j, grid, clock, fields, parameters, closure::Tuple{}, buoyancy) = zero(grid)
+
+@inline inner_top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple::Tuple{<:FlavorOfCATKE}, buoyancy) =
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[1], buoyancy)
+
+@inline top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple::Tuple{<:FlavorOfCATKE}, buoyancy) =
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[1], buoyancy)
+
+@inline top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple::Tuple{<:Any, <:Any}, buoyancy) =
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[1], buoyancy) + 
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[2], buoyancy)
+
+@inline inner_top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple::Tuple{<:Any, <:Any}, buoyancy) =
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[1], buoyancy) + 
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[2], buoyancy)
+
+@inline top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple::Tuple, buoyancy) =
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[1], buoyancy) + 
+    inner_top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[2:end], buoyancy)
+
+
+@inline inner_top_tke_flux(i, j, grid, clock, fields, parameters, closure::Tuple, buoyancy) =
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[1], buoyancy) +
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure_tuple[2:end], buoyancy) +
+=#
+
+@inline function _top_tke_flux(i, j, grid, tke::TurbulentKineticEnergyEquation, closure::CATKEVD,
+                               buoyancy, fields, top_tracer_bcs, top_velocity_bcs, clock)
+
+    wΔ³ = top_convective_turbulent_velocity³(i, j, grid, clock, fields, buoyancy, top_tracer_bcs)
+    u★ = friction_velocity(i, j, grid, clock, fields, top_velocity_bcs)
+
+    Cᵂu★ = tke.Cᵂu★
+    CᵂwΔ = tke.CᵂwΔ
+
+    return - Cᵂu★ * u★^3 - CᵂwΔ * wΔ³
+end
+
+""" Computes the friction velocity u★ based on fluxes of u and v. """
+@inline function friction_velocity(i, j, grid, clock, fields, velocity_bcs)
+    FT = eltype(grid)
+    Qᵘ = getbc(velocity_bcs.u, i, j, grid, clock, fields) 
+    Qᵛ = getbc(velocity_bcs.v, i, j, grid, clock, fields) 
+    return sqrt(sqrt(Qᵘ^2 + Qᵛ^2))
```

Comment:
```suggestion
    return (Qᵘ^2 + Qᵛ^2)^(FT(1/4))
```

```Julia
julia> A = rand(); B = rand();

julia> @time  sqrt(sqrt(A^2+B^2));
  0.000014 seconds (5 allocations: 80 bytes)

julia> @time  sqrt(sqrt(A^2+B^2));
  0.000014 seconds (5 allocations: 80 bytes)

julia> @time (A^2+B^2)^(Float64(1/4));
  0.000012 seconds (4 allocations: 64 bytes)

julia> @time (A^2+B^2)^(Float64(1/4));
  0.000010 seconds (4 allocations: 64 bytes)
```

Reply:
Hmm, your timings don't show a measurable difference. `btime` might though. Also we should do this on the GPU, that's where I expect a larger difference. Probably better to compute over an array too for the timing.

Reply:
As proof:

```julia
julia> A = rand(); B = rand();

julia> @time  sqrt(sqrt(A^2+B^2));
  0.000009 seconds (5 allocations: 80 bytes)

julia> @time  sqrt(sqrt(A^2+B^2));
  0.000009 seconds (5 allocations: 80 bytes)

julia> @time (A^2+B^2)^(Float64(1/4));
  0.000008 seconds (4 allocations: 64 bytes)

julia> @time (A^2+B^2)^(Float64(1/4));
  0.000010 seconds (4 allocations: 64 bytes)
```

Reply:
```julia
julia> A = rand(1024, 1024); B = rand(1024, 1024);

julia> @time @. (A^2+B^2)^(Float64(1/4));
  0.059826 seconds (14 allocations: 8.000 MiB)

julia> @time @. sqrt(sqrt(A^2+B^2));
  0.002154 seconds (16 allocations: 8.001 MiB)
```

Reply:
`sqrt` looks to be significantly faster than `pow(a, Float64)` on the CPU. Not sure about GPU though.

Reply:
OK!